File: PSP-DAM-ACTEVA03B Python/cocinero.py
import threading
import time

# Clase Cocinero: representa un hilo que prepara pedidos
class Cocinero(threading.Thread):
    def __init__(self, id, lista_pedidos):
        super().__init__()
        self.id = id
        self.lista_pedidos = lista_pedidos

    def run(self):
        while True:
            pedido = self.lista_pedidos.tomar_pedido()
            if pedido is None:
                break
            self.preparar_pedido(pedido)

    def preparar_pedido(self, pedido):
        print(f"Cocinero {self.id} está preparando el pedido {pedido.id}: {pedido.nombre_plato}")
        time.sleep(2)  # Simula el tiempo de preparación
        self.lista_pedidos.registrar_pedido(pedido, self.id)

File: PSP_DAM_ACTEVA03B Java/src/restaurante/Cocinero.java
package restaurante;

/**
 * Clase que extiende Thread para simular un cocinero
 * El cocinero toma pedidos de la lista y los prepara (simulado con sleep)
 */
public class Cocinero extends Thread {
    private int id;
    private ListaPedidos listaPedidos;

    public Cocinero(int id, ListaPedidos listaPedidos) {
        this.id = id;
        this.listaPedidos = listaPedidos;
    }

    @Override
    public void run() {
        while (true) {
            Pedido pedido = listaPedidos.tomarPedido();  // toma un pedido sincronizado
            if (pedido == null) {
                break;  // no quedan pedidos, termina hilo
            }
            prepararPedido(pedido);
        }
    }

    /**
     * Simula preparación con retardo y registra en listaPedidos
     */
    private void prepararPedido(Pedido pedido) {
        System.out.println("Cocinero " + id + " está preparando el pedido " +
                           pedido.getId() + ": " + pedido.getNombrePlato());
        try {
            Thread.sleep(2000); // 2 segundos, simulan preparación
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        listaPedidos.registrarPedido(pedido, id);  // registrar finalización
    }
}
